core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -102,6 +102,16 @@ sealed abstract class Aggregate extends Data {
 
   private[chisel3] def width: Width = getElements.map(_.width).foldLeft(0.W)(_ + _)
 
+  private[chisel3] def legacyAlwaysStrictConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
+    // If the source is a DontCare, generate a DefInvalid for the sink,
+    //  otherwise, issue a Connect.
+    if (that == DontCare) {
+      pushCommand(DefInvalid(sourceInfo, Node(this)))
+    } else {
+      pushCommand(Connect(sourceInfo, Node(this), Node(that)))
+    }
```

Comment:
Why are you directly using `Node` here instead of `.lref` which includes error detection for visibility?

Reply:
presumably because this is mostly a copy-paste of the stuff below? I will look into using the .lref instead...

Reply:
changing all these calls to use .lref the `ConnectSpec` and `CompatibilitySpec` all still pass